lib/spack/spack/installer.py

```diff
@@ -1418,6 +1427,40 @@ def _init_queue(self):
                 for dependent_id in dependents.difference(task.dependents):
                     task.add_dependent(dependent_id)
 
+    def _install_action(self, task):
+        """
+        Returns what kind of installation this task should perform.
```

Comment:
I think the following would be more clear:

```
"""
Determine whether the installation should be overwritten (if it already exists) or skipped (if has been handled by another process).

If the package has not been installed yet, this will indicate that the installation should proceed as normal (i.e. no need to transactionally preserve the old prefix).
"""
```